contrib/gitian-descriptors/assign_DISTNAME

```diff
@@ -0,0 +1,12 @@
+# Copyright (c) 2020 The Bitcoin Core developers
+# Distributed under the MIT software license, see the accompanying
+# file COPYING or http://www.opensource.org/licenses/mit-license.php.
+#
+# A helper script to be sourced into the gitian descriptors
+
+if RECENT_TAG="$(git describe --exact-match HEAD)"; then
+    VERSION="${RECENT_TAG#v}"
+else
+    VERSION="$(git rev-parse --short HEAD)"
```

Comment:
```suggestion
    VERSION="$(git rev-parse --short=10 HEAD)"
```

Not sure what the git logic is, but I couldn't verify from the doc that this is deterministic when the git version changes (and/or the number of objects in the repo)